xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
Ok, thanks for the explanation. Then how about removing the lock and making m_ThreadId atomic? This will make IsRunning() behave like most people expect it to while the impact on performance won't be noticeable. But if you think the code is correct as it is, I will drop this commit from the pull request. Your choice.